prow/flagutil/github.go

```diff
@@ -49,6 +53,7 @@ func (o *GitHubOptions) AddFlagsWithoutDefaultGitHubTokenPath(fs *flag.FlagSet)
 }
 
 func (o *GitHubOptions) addFlags(wantDefaultGitHubTokenPath bool, fs *flag.FlagSet) {
+	fs.StringVar(&o.host, "github-host", github.DefaultHost, "GitHub's default host (may differ for enterprise)")
 	o.endpoint = NewStrings(github.DefaultAPIEndpoint)
 	fs.Var(&o.endpoint, "github-endpoint", "GitHub's API endpoint (may differ for enterprise).")
```

Comment:
Shouldn't we be able to construct the REST and GraphQL API endoints from the host?

Reply:
I think that's a little bit tricky. If host is `github.com`, the endpoint of api is `api.github.com`, however for an enterprise github whose host is `foo.example.com`, its API endpoint is most probably going to be `foo.example.com/api/v3`. I'm ok to add that, but is it worth to do that(if public github, then X, otherwise Y)?

Reply:
I didn't know that the enterprise URLs would have different locations for the API. If that's the case, then no, what you have done here is better. We should just expose each endpoint separately.